youtube_dl/extractor/mdr.py

```diff
@@ -128,12 +140,19 @@ def _real_extract(self, url):
 
                     f = {
                         'url': video_url,
-                        'format_id': '%s-%d' % (media_type, vbr or abr),
-                        'filesize': filesize,
                         'abr': abr,
                         'preference': 1,
                     }
 
+                    if filesize:
+                        f.update({'filesize': filesize})
+
+                    if vbr or abr:
+                        f.update({
+                            'format_id': '%s-%d' % (media_type, vbr or abr)})
+                    else:
+                        f.update({'format_id': media_type})
```

Comment:
DRY.

Reply:
I guess my Python skills are inadequate for this. Is there a simpler way to do this?